### PR TITLE
Add ability to specify the scroll container node.

### PR DESCRIPTION
**Pull request by adamskoog**
I ran into an issue in my application where my scroll container is not the window. I made this change so I can pass in the node of my scroll container to get things working. Seems like something that may be useful to others that may come across your project.

### CODE REVIEW

**Comment by thcolin** [comment]
Thanks ! I'll get a look in the next days, I have an issue on an application too but didn't have time to fix it yet !

File: src/hooks/useVirtualGrid.js [hunk 1]
```diff
@@ -2,10 +2,10 @@ import useVirtualGridDisplay from './useVirtualGridDisplay'
 import useVirtualGridFirstRowIndex from './useVirtualGridFirstRowIndex'
 import useVirtualGridChildren from './useVirtualGridChildren'
 
-const useVirtualGrid = ({ cell, total, viewportRowOffset = 4, onRender }) => {
+const useVirtualGrid = ({ cell, total, viewportRowOffset = 4, onRender, scrollContainer }) => {
   const rowOffset = Math.max(2, Math.round(viewportRowOffset / 2) * 2)
   const { display, style, ref } = useVirtualGridDisplay({ cell, total, rowOffset })
-  const { firstRowIndex, scrolling } = useVirtualGridFirstRowIndex(display)
+  const { firstRowIndex, scrolling } = useVirtualGridFirstRowIndex({ ...display, scrollContainer })
```
Comment (thcolin) [inline src/hooks/useVirtualGrid.js]:
```suggestion
  const { firstRowIndex, scrolling } = useVirtualGridFirstRowIndex(display, scrollContainer)
```

File: src/hooks/useVirtualGridFirstRowIndex.js [hunk 2]
```diff
@@ -1,13 +1,22 @@
 import { useMemo, useEffect, useState, useRef } from 'react'
 import nanobounce from 'nanobounce'
 
-const useVirtualGridFirstRowIndex = ({ layout, cell, rowOffset }) => {
+const useVirtualGridFirstRowIndex = ({ layout, cell, rowOffset, scrollContainer }) => {
```
Comment (thcolin) [inline src/hooks/useVirtualGridFirstRowIndex.js]:
```suggestion
const useVirtualGridFirstRowIndex = ({ layout, cell, rowOffset }, scrollContainer) => {
```

File: src/hooks/useVirtualGridFirstRowIndex.js [hunk 3]
```diff
@@ -1,13 +1,22 @@
...
+  // Utilize the container that was passed in if so.
+  const container = scrollContainer ?? window;
+
   computeFirstRowIndex.current = () => {
-    const position = isClient ? Math.max(0, window.scrollY - layout.top) : 0
+
+    // window and divs use a different property for scroll
+    // position. Determine what we can get.
+    let scrollTop = window.scrollY;
+    if (container.scrollTop) scrollTop = container.scrollTop;
```
Comment (thcolin) [inline src/hooks/useVirtualGridFirstRowIndex.js]:
```suggestion
    const scrollTop = container === window ? container.scrollY : container.scrollTop
```

File: src/hooks/useVirtualGridFirstRowIndex.js [hunk 4]
```diff
@@ -1,13 +1,22 @@
...
 import nanobounce from 'nanobounce'
 
-const useVirtualGridFirstRowIndex = ({ layout, cell, rowOffset }) => {
+const useVirtualGridFirstRowIndex = ({ layout, cell, rowOffset, scrollContainer }) => {
   const isClient = typeof window === 'object'
   const computeFirstRowIndex = useRef()
   const debounce = useMemo(() => nanobounce(200), [])
 
+  // Utilize the container that was passed in if so.
+  const container = scrollContainer ?? window;
```
Comment (thcolin) [inline src/hooks/useVirtualGridFirstRowIndex.js]:
```suggestion
  const container = scrollContainer ?? window
```

**Review by thcolin — CHANGES_REQUESTED** [review]
Let me know if changes are ok for you

File: src/hooks/useVirtualGrid.js (hunk 1 above)
Comment (thcolin) [reply src/hooks/useVirtualGrid.js]: I prefer avoid any unecessary creation of new Object in this library for performances purpose

File: src/hooks/useVirtualGridFirstRowIndex.js (hunk 3 above)
Comment (thcolin) [reply src/hooks/useVirtualGridFirstRowIndex.js]: Just simplify with a one line to be coherent with code below

**Comment by thcolin** [comment]
👍

**Comment by adamskoog** [comment]
Glad I could help out!

**Comment by adamskoog** [comment]
I ran into a type issue with what I added. It seems like `PropTypes.node` doesn't do what I though.

I'm using useRef to get the reference to the scrollContainer, this seems to actually need `PropTypes.object` to not throw an error.

**Comment by thcolin** [comment]
Published in `0.0.20`, sorry for late fix !

**Pull request merged**